modules/packages/LinearAlgebra.chpl

```diff
@@ -806,6 +806,70 @@ proc inv (ref A: [?Adom] ?eltType, overwrite=false) where usingLAPACK {
   return A;
 }
 
+pragma "no doc"
+proc inv(ref A: [?Adom] ?eltType) throws
+where !usingLAPACK
+{
+  if Adom.rank != 2 then
+    halt("Wrong rank for matrix inverse");
+
+  if !isSquare(A) then
+    halt("Matrix inverse only supports square matrices");
```

Comment:
This can be a compilerError

Reply:
compilerError doesn't work if the argument to inv is a call to another function that returns a 2-D matrix. halt does

Reply:
That seems like a bug in the compiler to me... Do you have a short example demonstrating this?